### PR TITLE
[MyCollection] Wire up basic MP queries

### PR DESCRIPTION
**Pull request by damassi**
The type of this PR is: Enhancement 

### Description

This PR wires up basic My Collection graphql calls (list view, detail view). Still quite a bit of mock data in place, but this is a good building block.

Screen Shot 2020-08-19 at 5 08 00 PM

Screen Shot 2020-08-19 at 5 08 10 PM

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))

#trivial

### CODE REVIEW

File: src/lib/Scenes/Consignments/v2/Screens/ArtworkList/MyCollectionArtworkListItem.tsx [hunk 1]
```diff
@@ -1,27 +1,33 @@
...
-  artistNames: string
-  medium?: string
-  image?: {
-    url: string
-  }
-}
-
 interface MyCollectionArtworkListItemProps {
-  item: MyCollectionArtworkProps
+  artwork: MyCollectionArtworkListItem_artwork$key
```
Comment (damassi) [inline src/lib/Scenes/Consignments/v2/Screens/ArtworkList/MyCollectionArtworkListItem.tsx:10]: Note that this is a new in terms of previous patterns (the `$key` bit added to the variable name), and is required now that we're using relay hooks. A little bit annoying imo.

File: src/lib/Scenes/Consignments/v2/Screens/ArtworkDetail/MyCollectionArtworkDetail.tsx [hunk 2]
```diff
@@ -18,15 +20,51 @@ import { Insights } from "./Insights"
...
-  // Eventually this is happening via relay
-  const artwork = getArtworkByID(artworkID)
-  return <MyCollectionArtworkDetail artwork={artwork} />
-}
-
-export const MyCollectionArtworkDetail: React.FC<{ artwork: MyCollectionArtworkDetailArtwork }> = ({ artwork }) => {
+export const MyCollectionArtworkDetail: React.FC<{ artworkID: string }> = ({ artworkID }) => {
   const navActions = useStoreActions(actions => actions.navigation)
 
+  const { props, error } = useQuery<MyCollectionArtworkDetailQuery>(
```
Comment (damassi) [inline src/lib/Scenes/Consignments/v2/Screens/ArtworkDetail/MyCollectionArtworkDetail.tsx:26]: For top-level queries (equivalent to `QueryRenderer`) we need to pass the type into the `useQuery<T>` generic slot.

File: src/lib/Scenes/Consignments/v2/Screens/ArtworkDetail/MyCollectionArtworkDetail.tsx [hunk 3]
```diff
@@ -39,17 +77,31 @@ export const MyCollectionArtworkDetail: React.FC<{ artwork: MyCollectionArtworkD
           </Flex>
         </ScreenMargin>
 
-        <ArtworkImage artwork={artwork} />
+        <OpaqueImageView
+          // TODO: figure out if "normalized" is the correct version
+          imageURL={artwork.image.url.replace(":version", "normalized")}
+          height={200}
+          // TODO: see https://github.com/artsy/eigen/blob/master/src/lib/Containers/WorksForYou.tsx#L92 for getting the actual screen width
```
Comment (ds300) [inline src/lib/Scenes/Consignments/v2/Screens/ArtworkDetail/MyCollectionArtworkDetail.tsx:84]: can also go for `const {width} = useScreenDimensions()` if the image stretches the full width of the screen.

File: src/lib/Scenes/Consignments/v2/Screens/ArtworkDetail/MyCollectionArtworkDetail.tsx [hunk 4]
```diff
@@ -18,15 +20,51 @@ import { Insights } from "./Insights"
...
+    estimate: "$4,500 - $445,000",
+    image: {
+      url: "https://d32dm0rphc51dk.cloudfront.net/ng_LZVBhBb2805HMUIl6UQ/:version.jpg",
+    },
+    medium: "Pastels",
+    title: "Turtle",
+    hasInsights: true,
+    hasAuctionResults: true,
+    ...(props as any).artwork,
+  }
```
Comment (jonallured) [inline src/lib/Scenes/Consignments/v2/Screens/ArtworkDetail/MyCollectionArtworkDetail.tsx:66]: **question [non-blocking]:** fallbacks with override?

I think what I'm seeing here is a temporary construct that ensures the artwork const has all the things required to render the detail view. That last bit where we call `artwork` on `props` will cause whatever properties were returned from the server to override the fallback values, right?

*format via [Conventional Comments](https://conventionalcomments.org/)*

Comment (damassi) [reply src/lib/Scenes/Consignments/v2/Screens/ArtworkDetail/MyCollectionArtworkDetail.tsx:66]: Yup thats it

File: src/lib/Scenes/Consignments/v2/Screens/ArtworkDetail/MyCollectionArtworkDetail.tsx (hunk 3 above)
Comment (damassi) [reply src/lib/Scenes/Consignments/v2/Screens/ArtworkDetail/MyCollectionArtworkDetail.tsx:84]: 👍 will update

**Pull request merged**